release/serve_tests/workloads/single_deployment_1k_noop_replica.py

```diff
@@ -191,23 +188,51 @@ def main(num_replicas: Optional[int], num_trials: Optional[int],
     logger.info("Warming up cluster ....\n")
     warm_up_cluster(5, http_host, http_port)
 
-    final_result = []
+    avg_throughput = []
+    avg_latency = []
+    p50_latency = []
+    p75_latency = []
+    p90_latency = []
+    p99_latency = []
     for iteration in range(num_trials):
         logger.info(f"Starting wrk trial # {iteration + 1} ....\n")
-        # For detailed discussion, see https://github.com/wg/wrk/issues/205
-        # TODO:(jiaodong) What's the best number to use here ?
-        num_connections = int(num_replicas * DEFAULT_MAX_BATCH_SIZE * 0.75)
-        decoded_out = run_one_trial(trial_length, num_connections, http_host,
-                                    http_port)
-        metrics_dict = parse_wrk_decoded_stdout(decoded_out)
-        final_result.append(metrics_dict)
 
-    logger.info(f"Final results: {final_result}\n")
+        run_one_trial_remote = ray.remote(run_one_trial)
+
+        refs = []
+        for node in ray.nodes():
+            if node["Alive"]:
+                node_resource = f"node:{node['NodeManagerAddress']}"
+                refs.append(
+                    run_one_trial_remote.options(
+                        num_cpus=0, resources={
+                            node_resource: 0.01
+                        }).remote(trial_length, http_host, http_port))
+
+        for decoded_output in ray.get(refs):
+            parsed = parse_wrk_decoded_stdout(decoded_output)
+            avg_throughput.append(float(parsed["requests/sec"]))
+            avg_latency.append(float(parsed["latency_avg_ms"]))
+            p50_latency.append(float(parsed["P50_latency_ms"]))
+            p75_latency.append(float(parsed["P75_latency_ms"]))
+            p90_latency.append(float(parsed["P90_latency_ms"]))
+            p99_latency.append(float(parsed["P99_latency_ms"]))
+
+    final_results = {
+        "avg_throughput_qps": sum(avg_throughput) / len(avg_throughput),
+        "avg_latency_ms": sum(avg_latency) / len(avg_latency),
```

Comment:
i think this is also an approximation since each wrk trial we got have avg from different # of total requests, thus they have different weight on the avg latency reported ? Global avg latency should be more like 

for each wrk_result:
       total_latency += (avg_latency * # of requests)
       total_requests += # of requests 

avg_latency_ms = total_latency / total_requests

Reply:
Yep good point, take a note to fix this in follow-up?